src/main/isPhone.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class IsPhoneNumberConstraint implements ValidatorConstraintInterface {
+
+  validate (value: any, args: ValidationArguments): boolean {
+    if (value == null) {
+      return true
+    }
+
+    if (typeof value !== 'string') {
+      return false
+    }
+
+    value = value.replace(/\(|\)| |-|\+/g, '').replace(/^(00)?44/, '').replace(/^0/, '').replace(/[^0-9.]/g, '')
```

Comment:
It would be ideal to have each replace in new line. Can we reformat it please?

```
value.replace(/\(|\)| |-|\+/g, '')
  .replace(/^(00)?44/, '')
  .replace(/^0/, '')
  .replace(/[^0-9.]/g, '')
```